src/libmongoc/src/mongoc/mongoc-apm.c

```diff
@@ -981,7 +983,9 @@ mongoc_apm_is_sensitive_reply (const char *command_name, const bson_t *reply)
       return false;
    }
 
-   return bson_has_field (reply, "speculativeAuthenticate");
+   /* An empty command means it has been redacted */
+   return bson_count_keys (reply) == 0 ||
```

Comment:
Similar question about whether this check should be moved to `is_sensitive_command`.